docs: Governance docs Proposal (#1848)
llamastack/llama-stack: GOVERNANCE.md

*   Supporting new users (existing users are often the best people to help new users)
*   Creating, triaging or commenting on GitHub Issues
*   Doing code reviews or commenting on technical documents
*   Writing, editing, translating or reviewing the documentation
*   Organizing events or evangelizing the project (e.g,. on Social Media)

Contributors engage with the project through the issue tracker or by writing or editing documentation. They submit changes to the project itself via Pull Requests (PRs), which will be considered for inclusion in the project by existing Code Owners (see next section).

Contributors should follow the following guides when creating PRs:
- [Contribution Process](CONTRIBUTING.md)

As contributors gain experience and familiarity with the project, their profile and commitment within the community will
increase. At some stage, they may find themselves being nominated to become a Triager or Code Owner by an
existing Code Owner.

### Triagers
Triagers are contributors who have shown that they are committed to Llama Stack's continued development by helping to

> manage the project's backlog. They are responsible for ensuring that the project's issues are well-organized and up to

Reviewer: might be good to clarify what "manage" means

Reviewer 2: Added:

```
(track, label, assign issues - general issue hygiene) 
```

date so that contributors and Code Owners can navigate the project efficiently and prioritize key requests from

> the community.

Reviewer 2: addressed

### Code Owners

Code Owners (mapped in [CODEOWNERS](./.github/CODEOWNERS)) are community members who have shown that they are committed to Llama Stack's continued development through
ongoing engagement with the community. Because of this, they have the right to merge PRs and have voting rights.

> **Note**: Code Owners, like other contributors, must make changes to Llama Stack via pull requests (with code review).

> > This applies to all changes to documentation, code, configuration, governance, and so on.

Reviewer 2: removed it

- Triage GitHub issues, applying [labels](https://github.com/llamastack/llama-stack/labels) to each new item. Labels are extremely useful for future issue follow ups. Adding labels is somewhat subjective, so please use your best judgment.
- Make sure that ongoing PRs are moving forward at the right pace or closing them.

### Becoming a Code Owner

#### Who is eligible to become a Code Owner?
Anyone can become a Code Owner. Typically, a potential Code Owner will need to show that they understand the project,
its objectives, and its strategy. They will also have provided valuable contributions to the project over a period of
time. Code Owners must also act in the interest of the community.

#### Process
Any existing Code Owner can nominate new Code Owners by opening a PR to the [CODEOWNERS](./.github/CODEOWNERS) file. Once they have been nominated,
there will be a vote by the rest of the Code Owners. Code Owner voting is one of the few activities that takes place in
private. This is to allow Code Owners to express their opinions about a nominee without causing embarrassment freely.
The approval requires majority vote (>50%) from the Code Owners.

#### Earning a Nomination

> The nomination is at the discretion of the existing Code Owners and there is no single path of earning a nomination for,

Reviewer 2: Not necessarly.

*   You can start tackling issues labeled as ‘help wanted’, or if you are new to the project, some of the ‘good first issue’ tickets.
*   As you gain experience with the codebase and our standards, we will ask you to do code reviews for incoming PRs (i.e., all Code Owners are expected to shoulder a proportional share of community reviews).
*   We will expect you to start contributing increasingly complicated PRs, under the guidance of the existing Code Owners.

#### Losing Code Owner Status

If a Code Owner is no longer interested and cannot perform the Code Owner duties listed above, they can volunteer to be
moved to emeritus status. The Code Owner status is attributed for life otherwise. An emeritus Code Owner may request
reinstatement of commit access from the rest of Code Owners. Such reinstatement is subject to lazy consensus approval of
active Code Owners.

Emeritus status is a nominal title, and confers no special rights (like voting) or access. Emeritus members are
functionally identical to normal contributors, with the exception that they can request for reinstatement of their

> commit access.

Reviewer: maybe just say "codeowner access" - since access would be to discord channels, gitops workflows, in addition to commit access.

In extreme cases, Code Owners can lose their status by a vote of the Code Owners per the voting process below.

## Decision Making Process

Decisions about the future of Llama Stack are made through discussion with all community members, from the newest user

> to the most experienced maintainer. All non-sensitive project management discussion takes place on the project issue

Reviewer: ```suggestion
to the most experienced code owners. All non-sensitive project management discussion takes place on the project issue
```

tracker system. Occasionally, sensitive discussion occurs on a private channel of our Discord.

To ensure that the project is not bogged down by endless discussion and continual voting, the project operates a policy
of lazy consensus. This allows the majority of decisions to be made without resorting to a formal vote.

### Lazy consensus

Decision making typically involves the following steps:
*   Discussion *(in Discord channels and GitHub)*
*   Proposal *(via GitHub issue + GitHub PR)*

> *   (optional) Maintainers voting (if there is active opposition + consensus is not reached through discussion)

Reviewer: ```suggestion
*   (optional) Code Owners voting (if there is active opposition + consensus is not reached through discussion)
```

*   Decision

Any community member can make a proposal for consideration by the community. To initiate a discussion about a new idea,

> they should create an issue or submit a PR implementing the idea to the issue tracker. This will prompt a review and,

Reviewer: nit: what is "the issue tracker"? what does it mean to submit a PR to it?

Reviewer 2: removed.

there are no objections to the proposal. This requirement ensures that everyone is given enough time to read, digest,
and respond to the proposal. This time period is chosen to be as inclusive as possible of all participants, regardless
of their location and time commitments.

### Voting

Not all decisions can be made using lazy consensus. Issues such as those affecting the strategic direction or legal
standing of the project must gain explicit approval in the form of a vote. Every member of the community is encouraged
to express their opinions in all discussions and all votes. However, only project Code Owners have binding votes for
the purposes of decision making.

### Changes to Governance

We believe governance needs to adapt in order to be effective long term. This governance document itself can be extended
or modified as our community and project grows and our needs change.

A change in our governance structure should be a rare occurrence and should face sufficient scrutiny and review. To
this end, the rules that apply to modifications to the Llama Stack Governance structure are more stringent:

*   Governance changes are made through PRs to the repository.

> *   Lazy consensus applies to governance changes, but the proposed changes must be public for at least 7 days instead of 48 hours before they are accepted.

Reviewer 3: ```suggestion
*   Lazy consensus applies to governance changes, but the proposed changes must be public for at least 7 days instead of 72 hours before they are accepted.
```
